utils/heatmaper.py

```diff
@@ -0,0 +1,171 @@
+        super(Heatmaper, self).__init__()
+        # init cfg
+        self.cfg = cfg
+        # init device
+        self.device = device
+        # init model
+        self.model = model
+        # init transparency
+        self.alpha = alpha
+        # init filename prefix
+        self.prefix = prefix
+
+    def image_reader(self, image_file):
+        """
+        Args:
+            image_file: str to an image file path
+        Returns:
+            image: (1, C, H, W) tensor of image
+            image_color: (H, W, C) numpy array of BGR image
+        """
+        image_gray = cv2.imread(image_file, 0)
+        assert image_gray is not None, "invalid image read in: {}"\
+            .format(image_file)
+        image_color = cv2.cvtColor(image_gray, cv2.COLOR_GRAY2BGR)
+        image = transform(image_gray, self.cfg)
+        image = torch.from_numpy(image)
+        image = image.unsqueeze(0)
+        return image, image_color
+
+    def set_overlay(self, figure, row_, i, subtitle):
+        ax_overlay = figure.add_subplot(row_, 3, 2 + i)
+        ax_overlay.set_title(subtitle, fontsize=10, color='r')
+        ax_overlay.set_yticklabels([])
+        ax_overlay.set_xticklabels([])
+        ax_overlay.tick_params(axis='both', which='both', length=0)
+        return ax_overlay
+
+    def set_rawimage(self, figure, row_):
+        ax_raw_image = figure.add_subplot(row_, 3, 1)
+        subtitle = 'original image'
+        ax_raw_image.set_title(subtitle, fontsize=10, color='r')
+        ax_raw_image.set_yticklabels([])
+        ax_raw_image.set_xticklabels([])
+        ax_raw_image.tick_params(axis='both', which='both', length=0)
+        return ax_raw_image
+
+    def get_raw_image(self, image, ax):
+        """
+        Args:
+            image: (H, W, C) numpy array of BGR image
+            ax: AxesSubplot of matplotlib.axes
+        Returns:
+            image: AxesImage of matplotlib.image
+        """
+        image = ax.imshow(cv2.cvtColor(image, cv2.COLOR_BGR2RGB))
+        return image
+
+    def get_overlayed_smooth(self, ax, logit_map, prob_map):
+        long_side = self.cfg.long_side
+        image = ax.imshow(cv2.resize(prob_map, (long_side, long_side)),
+                          cmap='jet',
+                          vmin=0.0,
+                          vmax=1.0,
+                          alpha=self.alpha)
+        return image
+
+    def get_overlayed_img(self, ori_image, logit_map,
+                          prob_map, ax):
+        """
+        Args:
+            ori_image: (H, W) numpy array of gray image
+            logit_map: (H, W) numpy array of model prediction
+            prob_map: (H, W) numpy array of model prediction with prob
+            ax: AxesSubplot of matplotlib.axes
+        Returns:
+            image: AxesImage of matplotlib.image
+        """
+        overlayed_image = ax.imshow(ori_image, cmap='gray', vmin=0, vmax=255)
+        print(prob_map.shape)
```

Comment:
debug code?

Reply:
**TO ANSWER YOUR QUESTION**， This branch is not support for debug mode.
This is a redundant line.